gnd/src/main/java/com/google/android/gnd/persistence/local/room/AreaEntity.java

```diff
@@ -0,0 +1,149 @@
+import androidx.room.PrimaryKey;
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gms.maps.model.LatLngBounds;
+import com.google.android.gnd.model.basemap.Area;
+import com.google.auto.value.AutoValue;
+
+/** Represents a {@link Area} in the local data store. */
+@AutoValue
+@Entity(tableName = "area")
+public abstract class AreaEntity {
+  public static Area toArea(AreaEntity areaEntity) {
+    LatLng northEast = new LatLng(areaEntity.getNorth(), areaEntity.getEast());
+    LatLng southWest = new LatLng(areaEntity.getSouth(), areaEntity.getWest());
+    LatLngBounds bounds = new LatLngBounds(northEast, southWest);
+
+    Area.Builder area =
+        Area.newBuilder()
+            .setId(areaEntity.getId())
+            .setBounds(bounds)
+            .setState(toAreaState(areaEntity.getState()));
+    return area.build();
+  }
+
+  private static Area.State toAreaState(AreaEntityState state) {
+    switch (state) {
+      case PENDING:
+        return Area.State.PENDING;
+      case IN_PROGRESS:
+        return Area.State.IN_PROGRESS;
+      case DOWNLOADED:
+        return Area.State.DOWNLOADED;
+      case FAILED:
+        return Area.State.FAILED;
+      default:
+        throw new IllegalArgumentException("Unknown area state: " + state);
+    }
+  }
+
+  public static AreaEntity fromArea(Area area) {
+    AreaEntity.Builder entity =
+        AreaEntity.builder()
+            .setId(area.getId())
+            .setState(toEntityState(area.getState()))
+            .setNorth(area.getBounds().northeast.latitude)
+            .setEast(area.getBounds().northeast.longitude)
+            .setSouth(area.getBounds().southwest.latitude)
+            .setWest(area.getBounds().southwest.longitude);
+    return entity.build();
+  }
+
+  private static AreaEntityState toEntityState(Area.State state) {
+    switch (state) {
+      case PENDING:
+        return AreaEntityState.PENDING;
+      case IN_PROGRESS:
+        return AreaEntityState.IN_PROGRESS;
+      case FAILED:
+        return AreaEntityState.FAILED;
+      case DOWNLOADED:
+        return AreaEntityState.DOWNLOADED;
+      default:
+        return AreaEntityState.UNKNOWN;
+    }
+  }
+
+  public static AreaEntity create(String id, AreaEntityState state, Double north, Double east, Double south, Double west) {
+    return builder()
+        .setId(id)
+        .setState(state)
+        .setNorth(north)
+        .setEast(east)
+        .setSouth(south)
+        .setWest(west)
+        .build();
+  }
+
+  public static Builder builder() {
+    return new AutoValue_AreaEntity.Builder();
+  }
+
+  @AutoValue.CopyAnnotations
+  @NonNull
+  @PrimaryKey
+  @ColumnInfo(name = "id")
+  public abstract String getId();
+
+  @AutoValue.CopyAnnotations
+  @NonNull
+  @ColumnInfo(name = "state")
+  public abstract AreaEntityState getState();
+
+  @AutoValue.CopyAnnotations
+  @NonNull
+  @ColumnInfo(name = "north")
+  public abstract Double getNorth();
```

Comment:
Since these are not null, can we remove `@NonNull` and change them to `double` here and in the builder below?

Reply:
👍